docs/contributing/git-style.rst

```diff
@@ -46,3 +46,98 @@ features or plugins to streamline this process (e.g.,
 `magit <https://magit.vc/>`_). Git also provides ``git commit --fixup`` and
 ``git rebase --autosquash`` which can streamline this process.
 
+It is also possible to ask GitHub to squash your commits for you. This is the
+"squash and merge" option when merging your PR.
+
+Commit Message Style
+====================
+
+Commit messages are part of the repository in the same way as the files in a
+repository are. And as developers may use ``grep`` to orient themselves around
+the repository, developers may also use ``git log`` and ``git blame`` to
+understand more of the history of a file and its context. So, it pays to be
+mindful of the content and structure of commits and their messages.
+
+Commit messages should be written with your reviewers in mind as well as future
+developers (or generally the codebase itself). Changes to the repository are
+already included in a commit so use your commit message to focus on things that
+would not readily be apparent from just looking at the textual changes.
+
+Here are some examples of useful things to note for your reviewers and future
+developers:
+
+- Why was this change made?
+
+- What plausible alternative implementations did you consider but rule out?
+
+- What general thing or abstraction is this change in furtherance of?
+
+- If this change fixes a bug, what was the original bug or its symptoms?
+
+- If you did some additional changes along the way to facilitate the main
+  thrust of your work, explain the rational behind any non-obvious but required
+  changes.
+
+At the same time, there is no reason to agonize over the content of a commit
+message or be overly verbose. Applying an automatic formatter changes a lot of
+files but doesn't require a lot of context, while a one-line change might merit
+several paragraphs of context if the rationale is subtle or non-obvious.
+
+The subject line (i.e., the first line) of a commit message should be short,
+generally less than 80 characters long, and starts with a verb in the
+imperative mood (present tense, second person). You can think of the message as
```

Comment:
imperative mode. ;-)

Reply:
I'm not a linguist. I just followed wikipedia: https://en.wikipedia.org/wiki/Imperative_mood

FWIW, "imperative mode" has more hits in Google but they are mostly about non-English languages.

Reply:
.... Today I learned. That's super interesting. Carry on. And definitely check primary sources when I give you information apparently.

(I actually learned most of my grammar in the context of German, so this may have affected my thinking on this. It's not the only think, I still have a better idea of what the accusative is than what the indirect object is.)